src/js/containers/search/filters/recipient/RecipientNameDUNSContainer.jsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * RecipientNameDUNSContainer.jsx
+ * Created by michaelbray on 2/17/17.
+ */
+
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+
+import * as SearchHelper from 'helpers/searchHelper';
+import * as recipientActions from 'redux/actions/search/recipientActions';
+
+import Autocomplete from 'components/sharedComponents/autocomplete/Autocomplete';
+
+const propTypes = {
+    toggleRecipient: React.PropTypes.func,
+    setAutocompleteRecipients: React.PropTypes.func,
+    selectedRecipients: React.PropTypes.object,
+    autocompleteRecipients: React.PropTypes.array
+};
+
+export class RecipientNameDUNSContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            recipientSearchString: '',
+            autocompleteRecipients: []
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.timeout = null;
+    }
+
+    componentDidMount() {
+        this.parseAutocompleteRecipients(this.props.autocompleteRecipients);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!_.isEqual(nextProps.autocompleteRecipients, this.props.autocompleteRecipients)) {
+            this.parseAutocompleteRecipients(nextProps.autocompleteRecipients);
+        }
+    }
+
+    parseAutocompleteRecipients(recipients) {
+        const values = [];
+        if (recipients.length > 0) {
+            recipients.forEach((item) => {
+                values.push({
+                    title: item.recipient_name,
+                    subtitle: `DUNS: ${item.recipient_unique_id}`,
+                    data: item
+                });
+            });
+        }
+
+        this.setState({
+            autocompleteRecipients: values
+        });
+    }
+
+    queryAutocompleteRecipients(input) {
+        // Only search if input is 2 or more characters
+        if (input.length >= 2) {
+            this.setState({
+                recipientSearchString: input
+            });
+
+            if (this.recipientSearchRequest) {
+                // A request is currently in-flight, cancel it
+                this.recipientSearchRequest.cancel();
+            }
+
+            const recipientSearchParams = {
+                fields: ['recipient_name', 'recipient_unique_id'],
+                value: this.state.recipientSearchString,
+                matched_objects: true
+            };
+
+            this.recipientSearchRequest = SearchHelper.fetchRecipients(recipientSearchParams);
+
+            this.recipientSearchRequest.promise
+                .then((res) => {
+                    const data = res.data.matched_objects.recipient_name;
```

Comment:
This only returns recipients whose names matched the autocomplete string. DUNS matches are not included because the API puts those results in `recipient_unique_id`. Also, don't forget to remove duplicates that be returned by both fields.